src/pebble.py

```diff
@@ -276,6 +277,7 @@ def change_config(  # noqa: C901 pylint: disable=too-many-branches,too-many-stat
     """
     try:
         synapse.execute_migrate_config(container=container, charm_state=charm_state)
+        existing_synapse_config = _get_synapse_config(container)
         current_synapse_config = _get_synapse_config(container)
```

Comment:
I'm not sure I understand how these two values would ever be different - they're running the same things?

Reply:
Ok, I see how this works, you're passing current_synapse_config to enable_* methods below, which may modify it.